### ISSUE TITLE
core: batch periodic job summary incorrectly increments pending on each launch

### ISSUE THREAD

**Issue by jrasell** [hashicorp/nomad#9136]
### Nomad version
OSS master branch: `f6ca24c624707c807c7604048d7e24556d6a0853`

Behaviour seems to change (from correct to problematic) between `0.12.0` and `0.12.1` official release binaries.

### Operating system and Environment details
Reproduced locally on OSX as well as on Ubuntu 20.04 running in AWS.

### Issue
When an instance of a periodic job is launched by Nomad, it seems the job summary `Pending` value is incremented alongside the `Running` value. Once the job instance completes successfully and moves into the `dead` state, the `Pending` counter is left with its value incremented; the `Running` and `Dead` value update as expected.

### Reproduction steps
- Run Nomad in dev mode: `nomad agent -dev`
- Register the example job (detailed below): `nomad run <jobfile>`
- Once the periodic instance has been launched view the output of: `nomad job status batch-periodic-echo-env`
- Allow a few more instances of the job to launch, again view the job status output

```
ID                   = batch-periodic-echo-env
Name                 = batch-periodic-echo-env
Submit Date          = 2020-10-21T12:54:49+02:00
Type                 = batch
Priority             = 50
Datacenters          = dc1
Namespace            = default
Status               = running
Periodic             = true
Parameterized        = false
Next Periodic Launch = 2020-10-21T10:56:00Z (35s from now)

Children Job Summary
Pending  Running  Dead
3        0        3

Previously Launched Jobs
ID                                           Status
batch-periodic-echo-env/periodic-1603277700  dead
batch-periodic-echo-env/periodic-1603277880  dead
batch-periodic-echo-env/periodic-1603277940  dead
```

### Job file
```
job "batch-periodic-echo-env" {
  region      = "global"
  datacenters = ["dc1"]
  type        = "batch"

  periodic {
    cron             = "*/1 * * * * *"
    prohibit_overlap = true
  }

  group "echo-env" {
    count = 1

    task "echo-env" {
      driver = "docker"

      config {
        image  = "jrasell/echo-env:latest"
      }
    }
  }
}
```

### Nomad Server/Client logs (running in dev mode)
<details>
    <summary>logs</summary>

```shell
2020-10-21T12:35:18.184+0200 [DEBUG] nomad.periodic: registered periodic job: job="<ns: "default", id: "batch-periodic-echo-env">"
    2020-10-21T12:35:18.184+0200 [DEBUG] nomad.periodic: scheduled periodic job launch: launch_delay=41.815097s job="<ns: "default", id: "batch-periodic-echo-env">"
    2020-10-21T12:35:18.185+0200 [DEBUG] http: request complete: method=PUT path=/v1/jobs?region=global duration=2.175003ms
    2020-10-21T12:35:21.817+0200 [DEBUG] http: request complete: method=GET path=/v1/jobs duration=332.009µs
    2020-10-21T12:35:26.383+0200 [DEBUG] http: request complete: method=PUT path=/v1/search duration=420.868µs
    2020-10-21T12:35:26.470+0200 [DEBUG] http: request complete: method=GET path=/v1/jobs?prefix=batch-periodic-echo-env duration=295.17µs
    2020-10-21T12:35:26.471+0200 [DEBUG] http: request complete: method=GET path=/v1/job/batch-periodic-echo-env?namespace=default duration=301.558µs
    2020-10-21T12:35:26.473+0200 [DEBUG] http: request complete: method=GET path=/v1/job/batch-periodic-echo-env/summary?namespace=default duration=157.559µs
    2020-10-21T12:35:26.474+0200 [DEBUG] http: request complete: method=GET path=/v1/jobs?prefix=batch-periodic-echo-env%2Fperiodic- duration=102.032µs
    2020-10-21T12:36:00.003+0200 [DEBUG] nomad.periodic:  launching job: job="<ns: "default", id: "batch-periodic-echo-env">" launch_time="2020-10-21 10:36:00 +0000 UTC"
    2020-10-21T12:36:00.003+0200 [DEBUG] nomad.periodic: scheduled periodic job launch: launch_delay=59.996192s job="<ns: "default", id: "batch-periodic-echo-env">"
    2020-10-21T12:36:00.003+0200 [DEBUG] worker: dequeued evaluation: eval_id=783922b9-1465-cf68-5111-d08987fb9bf9
    2020-10-21T12:36:00.004+0200 [DEBUG] worker.batch_sched: reconciled current state with desired state: eval_id=783922b9-1465-cf68-5111-d08987fb9bf9 job_id=batch-periodic-echo-env/periodic-1603276560 namespace=default results="Total changes: (place 1) (destructive 0) (inplace 0) (stop 0)
Desired Changes for "echo-env": (place 1) (inplace 0) (destructive 0) (stop 0) (migrate 0) (ignore 0) (canary 0)"
    2020-10-21T12:36:00.005+0200 [DEBUG] worker: submitted plan for evaluation: eval_id=783922b9-1465-cf68-5111-d08987fb9bf9
    2020-10-21T12:36:00.005+0200 [DEBUG] worker.batch_sched: setting eval status: eval_id=783922b9-1465-cf68-5111-d08987fb9bf9 job_id=batch-periodic-echo-env/periodic-1603276560 namespace=default status=complete
    2020-10-21T12:36:00.005+0200 [DEBUG] worker: updated evaluation: eval="<Eval "783922b9-1465-cf68-5111-d08987fb9bf9" JobID: "batch-periodic-echo-env/periodic-1603276560" Namespace: "default">"
    2020-10-21T12:36:00.005+0200 [DEBUG] worker: ack evaluation: eval_id=783922b9-1465-cf68-5111-d08987fb9bf9
    2020-10-21T12:36:00.005+0200 [DEBUG] client: updated allocations: index=14 total=1 pulled=1 filtered=0
    2020-10-21T12:36:00.005+0200 [DEBUG] client: allocation updates: added=1 removed=0 updated=0 ignored=0
    2020-10-21T12:36:00.006+0200 [DEBUG] client: allocation updates applied: added=1 removed=0 updated=0 ignored=0 errors=0
    2020-10-21T12:36:00.007+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: starting plugin: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad args=[/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad, logmon]
    2020-10-21T12:36:00.010+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: plugin started: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad pid=72181
    2020-10-21T12:36:00.010+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: waiting for RPC address: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad
    2020-10-21T12:36:00.022+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon.nomad: plugin address: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env network=unix @module=logmon address=/var/folders/sx/83shrk2s7bs09xvn51vpsxkr0000gn/T/plugin316938392 timestamp=2020-10-21T12:36:00.022+0200
    2020-10-21T12:36:00.022+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: using plugin: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env version=2
    2020-10-21T12:36:00.024+0200 [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env @module=logmon path=/private/var/folders/sx/83shrk2s7bs09xvn51vpsxkr0000gn/T/NomadClient507233102/d1a1f70d-b998-2fac-f998-d4bbebc8e2f1/alloc/logs/.echo-env.stdout.fifo timestamp=2020-10-21T12:36:00.024+0200
    2020-10-21T12:36:00.024+0200 [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad: opening fifo: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env path=/private/var/folders/sx/83shrk2s7bs09xvn51vpsxkr0000gn/T/NomadClient507233102/d1a1f70d-b998-2fac-f998-d4bbebc8e2f1/alloc/logs/.echo-env.stderr.fifo @module=logmon timestamp=2020-10-21T12:36:00.024+0200
    2020-10-21T12:36:00.176+0200 [DEBUG] client: updated allocations: index=16 total=1 pulled=0 filtered=1
    2020-10-21T12:36:00.176+0200 [DEBUG] client: allocation updates: added=0 removed=0 updated=0 ignored=1
    2020-10-21T12:36:00.176+0200 [DEBUG] client: allocation updates applied: added=0 removed=0 updated=0 ignored=1 errors=0
    2020-10-21T12:36:01.598+0200 [DEBUG] client.driver_mgr.docker: docker pull succeeded: driver=docker image_ref=jrasell/echo-env:latest
    2020-10-21T12:36:01.601+0200 [DEBUG] client.driver_mgr.docker: image reference count incremented: driver=docker image_name=jrasell/echo-env:latest image_id=sha256:4719dbe58f5f27d90aef1a52106a60dd2a4ef79b9857a8b3837a3a2918a91768 references=1
    2020-10-21T12:36:01.601+0200 [DEBUG] client.driver_mgr.docker: configured resources: driver=docker task_name=echo-env memory=314572800 memory_reservation=0 cpu_shares=100 cpu_quota=0 cpu_period=0
    2020-10-21T12:36:01.601+0200 [DEBUG] client.driver_mgr.docker: binding directories: driver=docker task_name=echo-env binds="[]string{"/private/var/folders/sx/83shrk2s7bs09xvn51vpsxkr0000gn/T/NomadClient507233102/d1a1f70d-b998-2fac-f998-d4bbebc8e2f1/alloc:/alloc", "/private/var/folders/sx/83shrk2s7bs09xvn51vpsxkr0000gn/T/NomadClient507233102/d1a1f70d-b998-2fac-f998-d4bbebc8e2f1/echo-env/local:/local", "/private/var/folders/sx/83shrk2s7bs09xvn51vpsxkr0000gn/T/NomadClient507233102/d1a1f70d-b998-2fac-f998-d4bbebc8e2f1/echo-env/secrets:/secrets"}"
    2020-10-21T12:36:01.601+0200 [DEBUG] client.driver_mgr.docker: networking mode not specified; using default: driver=docker task_name=echo-env
    2020-10-21T12:36:01.601+0200 [DEBUG] client.driver_mgr.docker: applied labels on the container: driver=docker task_name=echo-env labels=map[com.hashicorp.nomad.alloc_id:d1a1f70d-b998-2fac-f998-d4bbebc8e2f1]
    2020-10-21T12:36:01.601+0200 [DEBUG] client.driver_mgr.docker: setting container name: driver=docker task_name=echo-env container_name=echo-env-d1a1f70d-b998-2fac-f998-d4bbebc8e2f1
    2020-10-21T12:36:01.728+0200 [INFO]  client.driver_mgr.docker: created container: driver=docker container_id=b9a85ec91b6d17eb49bbaa0b98574208dcbbf7b5b8ea35b4b6b17bfb73b321f2
    2020-10-21T12:36:06.943+0200 [DEBUG] http: request complete: method=PUT path=/v1/search duration=557.959µs
    2020-10-21T12:36:07.022+0200 [DEBUG] http: request complete: method=GET path=/v1/jobs?prefix=batch-periodic-echo-env duration=431.171µs
    2020-10-21T12:36:07.023+0200 [DEBUG] http: request complete: method=GET path=/v1/job/batch-periodic-echo-env?namespace=default duration=125.549µs
    2020-10-21T12:36:07.026+0200 [DEBUG] http: request complete: method=GET path=/v1/job/batch-periodic-echo-env/summary?namespace=default duration=106.295µs
    2020-10-21T12:36:07.026+0200 [DEBUG] http: request complete: method=GET path=/v1/jobs?prefix=batch-periodic-echo-env%2Fperiodic- duration=114.842µs
    2020-10-21T12:36:12.555+0200 [INFO]  client.driver_mgr.docker: started container: driver=docker container_id=b9a85ec91b6d17eb49bbaa0b98574208dcbbf7b5b8ea35b4b6b17bfb73b321f2
    2020-10-21T12:36:12.555+0200 [DEBUG] client.driver_mgr.docker.docker_logger: starting plugin: driver=docker path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad args=[/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad, docker_logger]
    2020-10-21T12:36:12.558+0200 [DEBUG] client.driver_mgr.docker.docker_logger: plugin started: driver=docker path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad pid=72207
    2020-10-21T12:36:12.558+0200 [DEBUG] client.driver_mgr.docker.docker_logger: waiting for RPC address: driver=docker path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad
    2020-10-21T12:36:12.569+0200 [DEBUG] client.driver_mgr.docker.docker_logger.nomad: plugin address: driver=docker network=unix @module=docker_logger address=/var/folders/sx/83shrk2s7bs09xvn51vpsxkr0000gn/T/plugin418798746 timestamp=2020-10-21T12:36:12.569+0200
    2020-10-21T12:36:12.569+0200 [DEBUG] client.driver_mgr.docker.docker_logger: using plugin: driver=docker version=2
    2020-10-21T12:36:12.570+0200 [DEBUG] client.driver_mgr.docker.docker_logger.nomad: using client connection initialized from environment: driver=docker @module=docker_logger timestamp=2020-10-21T12:36:12.570+0200
    2020-10-21T12:36:12.750+0200 [DEBUG] client.driver_mgr.docker.docker_logger: plugin process exited: driver=docker path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad pid=72207
    2020-10-21T12:36:12.750+0200 [DEBUG] client.driver_mgr.docker.docker_logger: plugin exited: driver=docker
    2020-10-21T12:36:12.753+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.stats_hook: failed to start stats collection for task with unrecoverable error: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env error="container stopped"
    2020-10-21T12:36:12.762+0200 [DEBUG] client.driver_mgr.docker: image id reference count decremented: driver=docker image_id=sha256:4719dbe58f5f27d90aef1a52106a60dd2a4ef79b9857a8b3837a3a2918a91768 references=0
    2020-10-21T12:36:12.762+0200 [INFO]  client.alloc_runner.task_runner: not restarting task: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env reason="Restart unnecessary as task terminated successfully"
    2020-10-21T12:36:12.762+0200 [INFO]  client.gc: marking allocation for GC: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1
    2020-10-21T12:36:12.764+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: plugin process exited: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env path=/Users/jrasell/go/src/github.com/hashicorp/nomad/bin/nomad pid=72181
    2020-10-21T12:36:12.764+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: plugin exited: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env
    2020-10-21T12:36:12.764+0200 [DEBUG] client.alloc_runner.task_runner: task run loop exiting: alloc_id=d1a1f70d-b998-2fac-f998-d4bbebc8e2f1 task=echo-env
```

</details>

**Comment by jrasell** [comment]
closing in favour of https://github.com/hashicorp/nomad/issues/8903
